nomad/core_sched_test.go

```diff
@@ -1928,6 +1939,19 @@ func TestAllocation_GCEligible(t *testing.T) {
 			NextAllocID: uuid.Generate(),
 			ShouldGC:    true,
 		},
+		{
```

Comment:
Can we add one where it does GC because Next is set

Reply:
that's already there in line 1928